internal/zipper/zipper_test.go

```diff
@@ -0,0 +1,130 @@
+package zipper
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/composite-action-framework-go/pkg/fs"
+)
+
+type files map[string]string
+
+func TestZipper_ZipDir_ok(t *testing.T) {
+
+	cases := []files{
+		{
+			"text.txt": "hello!",
+		},
+		{
+			"subdir/text.txt": "hello!",
+		},
+		{
+			"subdir/text.txt":         "hello!",
+			"subdir/blah/blah/hi.txt": "hello!",
+		},
+	}
+
+	for _, c := range cases {
+		inputFiles := c
+		t.Run("", func(t *testing.T) {
+			buf := &bytes.Buffer{}
+			z := New(buf)
+
+			dir := createTestDir(t, inputFiles)
+			if err := z.ZipDir(dir); err != nil {
+				t.Fatal(err)
+			}
+
+			byteReader := bytes.NewReader(buf.Bytes())
+			reader, err := zip.NewReader(byteReader, int64(buf.Len()))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			wantFiles := make(files, len(inputFiles))
+			for name := range inputFiles {
+				// We expect to see a flattened hierarchy here,
+				// so throw away the dir component.
+				wantFiles[filepath.Base(name)] = ""
```

Comment:
Only the keys here are important as we're just checking that the zip contains the expected filenames. We could expand the check to cover file contents as well if that seems useful...